### PR TITLE
Refactored integration tests

### PR DESCRIPTION
**Pull request by yuchaoran2011**
* Added an integration test for mounting a ConfigMap. This is a common scenario that we used to check manually. Currently webhook is required. But the test can be adapted with minimal changes once pod template is supported in the operator.
* Many minor bug fixes and improvements in the test code.
* Upgraded Kubernetes Go client version to 1.13.4.
* Upgraded dep to 0.5.1

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -75,19 +75,19 @@ required = [
 
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.11.8"
+  version = "kubernetes-1.13.4"
```
Comment (liyinan926) [inline Gopkg.toml]: Why do we want to upgrade to this version?

File: examples/spark-pi-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+apiVersion: "sparkoperator.k8s.io/v1beta1"
+kind: SparkApplication
+metadata:
+  name: spark-pi
+  namespace: default
+spec:
+  type: Scala
+  mode: cluster
+  image: "lightbend/spark:2.1.0-OpenShift-2.4.0-rh"
```
Comment (liyinan926) [inline examples/spark-pi-configmap.yaml]: This needs to be changed.

File: examples/spark-pi.yaml [hunk 3]
```diff
@@ -22,7 +21,7 @@ metadata:
 spec:
   type: Scala
   mode: cluster
-  image: "gcr.io/spark-operator/spark:v2.4.0"
+  image: "lightbend/spark:2.1.0-OpenShift-2.4.0-rh"
```
Comment (liyinan926) [inline examples/spark-pi.yaml]: Ditto.

File: sparkctl/cmd/event.go [hunk 4]
```diff
@@ -151,7 +153,9 @@ func streamEvents(events watch.Interface, streamSince int64) error {
 		// Start rendering contents of the table without table header as it is already printed
 		table = prepareNewTable()
 		table.SetBorders(tablewriter.Border{Left: true, Top: false, Right: true, Bottom: false})
-		_, err := watch.Until(watchExpire, events, func(ev watch.Event) (bool, error) {
+		ctx := context.Background()
```
Comment (liyinan926) [inline sparkctl/cmd/event.go]: Let's use a TODO instead.

File: manifest/spark-operator-with-webhook.yaml [hunk 5]
```diff
@@ -78,7 +79,7 @@ spec:
       - name: main
         image: gcr.io/spark-operator/spark-operator:v2.4.0-v1beta1-latest
         imagePullPolicy: IfNotPresent
-        command: ["/usr/bin/gencerts.sh", "-p"]
+        command: ["/usr/bin/gencerts.sh", "-n", "spark-operator", "-s", "spark-webhook", "-p"]
```
Comment (liyinan926) [inline manifest/spark-operator-with-webhook.yaml]: Do we need this change?

File: Gopkg.toml (hunk 1 above)
Comment (yuchaoran2011) [reply Gopkg.toml]: There are multiple Kubernetes Go client APIs that have been removed/changed in the new version. Examples include `WorkQueueMetrics` and `Until` that got changed in this PR. And in general, we think it's good to bring the API version used by the operator to a more recent one.

Comment (liyinan926) [reply Gopkg.toml]: OK, makes sense.

File: manifest/spark-operator-with-webhook.yaml (hunk 5 above)
Comment (yuchaoran2011) [reply manifest/spark-operator-with-webhook.yaml]: You are right. I forgot that they are already defaults in `gencerts.sh`. I've undo this change.

File: sparkctl/cmd/event.go (hunk 4 above)
Comment (yuchaoran2011) [reply sparkctl/cmd/event.go]: Done.

**Pull request merged**